spec/model/implementation/effective-module-specification.spec.ts

```diff
@@ -0,0 +1,70 @@
+import {
+    EffectiveModuleSpecification,
+    EffectiveModuleSpecificationClause,
+} from '../../../src/model/implementation/modules/effective-module-specification';
+import { expect } from 'chai';
+
+describe('EffectiveModuleSpecification', () => {
+    describe('simplify()', () => {
+        it('removes duplicate simple clauses', () => {
+            const input = new EffectiveModuleSpecification({
+                orCombinedClauses: [
+                    new EffectiveModuleSpecificationClause({ andCombinedModules: ['a'] }),
+                    new EffectiveModuleSpecificationClause({ andCombinedModules: ['b'] }),
+                    new EffectiveModuleSpecificationClause({ andCombinedModules: ['a'] }),
+                ],
+            });
+            const output = input.simplify();
+            expect(output.toString()).to.equal('a, b');
+        });
+
+        it('removes duplicate multi-module clauses', () => {
+            const input = new EffectiveModuleSpecification({
+                orCombinedClauses: [
+                    new EffectiveModuleSpecificationClause({ andCombinedModules: ['a', 'c'] }),
+                    new EffectiveModuleSpecificationClause({ andCombinedModules: ['b'] }),
+                    new EffectiveModuleSpecificationClause({ andCombinedModules: ['a', 'c'] }),
+                ],
+            });
+            const output = input.simplify();
+            expect(output.toString()).to.equal('b, a && c');
+        });
+
+        it('removes redundant clauses', () => {
+            const input = new EffectiveModuleSpecification({
+                orCombinedClauses: [
+                    new EffectiveModuleSpecificationClause({ andCombinedModules: ['a', 'c'] }),
+                    new EffectiveModuleSpecificationClause({ andCombinedModules: ['b'] }),
+                    new EffectiveModuleSpecificationClause({ andCombinedModules: ['a', 'c', 'd'] }),
+                ],
+            });
+            const output = input.simplify();
+            expect(output.toString()).to.equal('b, a && c');
+        });
```

Comment:
Why is `d` redundant in this case?

Shouldn't it be `b, d, a && c` ?

Reply:
if the field is already required by the combination of modules "a" and "c", it's implied that the field is required if you have "a", "c" and "d" enabled.

Reply:
Ah 💡 , yep

Reply:
Sorry, forgot to check your comment and approve the MR yesterday.